src/session_pool/session_pool.js

```diff
@@ -228,15 +261,17 @@ export class SessionPool extends EventEmitter {
     }
 
     /**
-     * Removes `Session` instance from `SessionPool`.
-     * @param {Session} session  - Session to be removed
+     * Removes retired `Session` instances from `SessionPool`.
      * @private
      */
-    _removeSession(session) {
-        const sessionIndex = this.sessions.findIndex((storedSession) => storedSession.id === session.id);
+    _removeRetiredSessions() {
+        this.sessions = this.sessions.filter((storedSession) => {
+            const sessionIsUsable = storedSession.isUsable();
+            if (sessionIsUsable) return storedSession;
```

Comment:
this is `filter`, you return `true` or `false`, not objects.

Reply:
Same comment as above. I would still leave explicit `if (storedSession.isUsable()) return true;` as need to remove retired sessions from Map